reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscription.scala

```diff
@@ -68,14 +68,17 @@ private[reactivestreams] final class StreamSubscription[F[_], A] private (
         go(in).stream
       }
 
-    stream
-      .through(subscriptionPipe)
-      .interruptWhen(canceled)
-      .evalMap(x => F.delay(sub.onNext(x)))
-      .handleErrorWith(e => Stream.eval(onError(e)))
-      .onFinalize(canceled.get.ifM(ifTrue = F.unit, ifFalse = onComplete))
-      .compile
-      .drain
+    val prog =
+      stream
+        .through(subscriptionPipe)
+        .interruptWhen(canceled)
+        .evalMap(x => F.delay(sub.onNext(x)))
+        .handleErrorWith(e => Stream.eval(onError(e)))
+        .onFinalize(canceled.get.ifM(ifTrue = F.unit, ifFalse = onComplete))
+        .compile
+        .drain
+
+    F.onCancel(prog, cancelMe)
```

Comment:
I wonder if this should be a `.guarantee(...)`, since we probably want to set canceled no matter how we complete? I also wonder if it's redundant with the `onFinalize`.

Reply:
Yeah is redundant.